services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -156,6 +160,68 @@ class SwitchRulesSpec extends BaseSpecification {
         ]
     }
 
+    @Unroll
+    def "Able to delete switch rules by #data.description"() {
+        given: "A switch with some flow rules installed"
+        northboundService.addFlow(flow)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northboundService.getFlowStatus(flow.id).status == FlowState.UP
+            assert northboundService.getSwitchRules(data.switch.dpId).flowEntries.size() ==
+                    defaultRules.size() + flowRulesCount
+        }
+
+        when: "Delete switch rules by #data.description"
+        def deletedRules = northboundService.deleteSwitchRules(data.switch.dpId, null, data.inPort, data.inVlan,
+                data.outPort)
+
+        then: "The requested rules are really deleted"
+        deletedRules.size() == data.rulesDeleted
+        Wrappers.wait(RULES_DELETION_TIME) {
+            def actualRules = northboundService.getSwitchRules(data.switch.dpId).flowEntries
+            assert actualRules.size() == data.rulesRemained
+            assert filterRules(actualRules, data.inPort, data.inVlan, data.outPort).empty
+        }
+
+        and: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        where:
+        flow << [buildFlow()] * 4
+        data << [[description  : "inPort",
+                  switch       : srcSwitch,
+                  inPort       : flow.source.portNumber,
+                  inVlan       : null,
+                  outPort      : null,
+                  rulesDeleted : 1,
+                  rulesRemained: defaultRules.size() + flowRulesCount - 1
```

Comment:
rulesDeleted and rulesRemained don't seem to be changing between iterations. It seems that you can add negative cases here by feeding non-existing port-vlans and changing these params, can't you?

Reply:
No I don't insist, but I find the less code the better. Otherwise, if you are not planning to extend this, then you can probably remove params that are static

Reply:
I agree, less code is better. But positive tests are positive. Negative tests are negative. Yep, I will delete static values.